torch/_inductor/pattern_matcher.py

```diff
@@ -598,6 +598,224 @@ def addmm(match, mat1, mat2, inp):
+            return torch.ops.mkldnn._convolution_pointwise(
+                input,
+                weight,
+                bias,
+                padding,
+                stride,
+                dilation,
+                groups,
+                unary_op.op_name,
+                unary_op.scalars_attr,
+                unary_op.algorithm_attr,
+            )
+
+        return fn
+
+    for unary_op, patterns in replacement_unary_fusion_patterns.items():
+        register_mkldnn_conv_replacement_pattern(unary_op, patterns[0])
+        # TODO: add linear/ConvTranspose fusion
+
+    @register_lowering_pattern(CallFunction(aten.mkldnn_convolution, *_conv_args))
+    def single_conv_lowering(
+        match, input, weight, bias, padding, stride, dilation, groups
+    ):
+        return L[torch.ops.mkldnn._convolution_pointwise](
+            input, weight, bias, padding, stride, dilation, groups, "none", [], ""
+        )
+
+    def register_leaky_relu_fusion_lowering(computation_call, computation_op):
+        @register_lowering_pattern(leaky_relu_fusion(computation_call))
+        def fn(match, *args, **kwargs):
+            negative_slope = kwargs.get("negative_slope")
+            if isinstance(negative_slope, ir.TensorBox):
+                matched = False
+            else:  # inp is a Number
+                matched = True
+            computation_args = [arg for arg in args]
+            if matched:
+                computation_args += ["leaky_relu", [negative_slope], ""]
+                return L[computation_op](*computation_args)
+            else:
+                computation_args += ["none", [], ""]
+                computation_out = L[computation_op](*computation_args)
+                return L[aten.where](
+                    L[aten.gt](computation_out, 0),
+                    computation_out,
+                    L[aten.mul](computation_out, negative_slope),
+                )
+
+        return fn
+
+    def register_hardtanh_fusion_lowering(computation_call, computation_op):
+        @register_lowering_pattern(hardtanh_fusion(computation_call))
+        def fn(match, *args, **kwargs):
+            min_value = kwargs.get("min_value")
+            max_value = kwargs.get("max_value")
+            if isinstance(min_value, ir.TensorBox) or isinstance(
+                max_value, ir.TensorBox
+            ):
+                matched = False
+            else:  # inp is a Number
+                matched = True
+            computation_args = [arg for arg in args]
+            if matched:
+                computation_args += ["hardtanh", [min_value, max_value], ""]
+                return L[computation_op](*computation_args)
+            else:
+                computation_args += ["none", [], ""]
+                conv_out = L[computation_op](*computation_args)
+                return L[aten.clamp_max](
+                    L[aten.clamp_min](conv_out, min_value), max_value
+                )
```

Comment:
This is the decompose version of `hardtanh`, right? Is there any way to call the decomposed function directly but not duplicate it again.

Reply:
I didn't find a way to do it.